schemas/v1/8_vm_messages.go

```diff
@@ -0,0 +1,46 @@
+package v1
+
+func init() {
+	patches.Register(
+		8,
+		`
+ CREATE TABLE {{ .SchemaName | default "public"}}.vm_messages (
+    height bigint NOT NULL,
+    state_root text NOT NULL,
+    cid text NOT NULL,
+	source text,
+    "from" text NOT NULL,
+    "to" text NOT NULL,
+    value numeric NOT NULL,
+    method bigint NOT NULL,
+    actor_code text NOT NULL,
+    exit_code bigint NOT NULL,
+    gas_used bigint NOT NULL,
+    params jsonb,
+	returns jsonb
+);
+ALTER TABLE ONLY {{ .SchemaName | default "public"}}.vm_messages ADD CONSTRAINT vm_messages_pkey PRIMARY KEY (height, state_root, cid, source);
+CREATE INDEX vm_messages_height_idx ON {{ .SchemaName | default "public"}}.vm_messages USING BTREE (height);
+CREATE INDEX vm_messages_cid_idx ON {{ .SchemaName | default "public"}}.vm_messages USING HASH (cid);
+CREATE INDEX vm_messages_source_idx ON {{ .SchemaName | default "public"}}.vm_messages USING HASH (source);
+CREATE INDEX vm_messages_from_idx ON {{ .SchemaName | default "public"}}.vm_messages USING HASH ("from");
+CREATE INDEX vm_messages_to_idx ON {{ .SchemaName | default "public"}}.vm_messages USING HASH ("to");
+CREATE INDEX vm_messages_actor_code_method_idx ON {{ .SchemaName | default "public"}}.vm_messages USING BTREE (actor_code, method);
```

Comment:
If we want to do a write-optimized schema, I'd recommend to get rid of several of these indices since all these indices need to be updated whenever an `insert` or `update` happens.

I propose the following:

```suggestion
CREATE INDEX vm_messages_height_idx ON {{ .SchemaName | default "public"}}.vm_messages USING BTREE (height);
CREATE INDEX vm_messages_from_idx ON {{ .SchemaName | default "public"}}.vm_messages USING HASH ("from");
CREATE INDEX vm_messages_to_idx ON {{ .SchemaName | default "public"}}.vm_messages USING HASH ("to");
```

Regarding: 
```sql
CREATE INDEX vm_messages_actor_code_method_idx ON {{ .SchemaName | default "public"}}.vm_messages USING BTREE (actor_code, method);
```
This only makes sense if we have queries that have a filter that's `where actor_code = x and method = b` or `where actor_code = x` . This wouldn't work well with filters: `where method = b and actor_code = x` or `where method = b`. 

Therefore, I'd recommend based on Mike's [comment here](https://github.com/filecoin-project/lily/issues/978#issuecomment-1203227156):

```sql
CREATE INDEX vm_messages_actor_code_idx ON {{ .SchemaName | default "public"}}.vm_messages USING BTREE (actor_code);
```

Reply:
> This only makes sense if we have queries that have a filter that's where actor_code = x and method = b

This will be a common query since the method number depends on the actor code. i.e. method numbers only have meaning when compared with the actor they were called on.